tests/transformers/test_transformers_model_export.py

```diff
@@ -1919,3 +1922,62 @@ def test_parse_list_output_for_multiple_candidate_pipelines(mock_pyfunc_wrapper)
         mock_pyfunc_wrapper._parse_list_output_for_multiple_candidate_pipelines(output_data)
         == expected_output
     )
+
+
+@pytest.mark.parametrize(
+    "pipeline_input, pipeline_output, expected_output",
+    [
+        (
+            "What answers?",
+            [{"generated_text": "What answers?\n\nA collection of \n\nanswers"}],
+            "A collection of answers",
+        ),
+        ("Hello!", [{"generated_text": "Hello!\n\nHow are you?"}], "How are you?"),
+        (
+            ["Hi!", "What's up?"],
+            [[{"generated_text": "Hi!\nHello there"}, {"generated_text": "Not much, and you?"}]],
+            ["Hello there", "Not much, and you?"],
+        ),
+    ],
+)
+def test_parse_input_from_instruction_pipeline(
+    mock_pyfunc_wrapper, pipeline_input, pipeline_output, expected_output
+):
+    assert (
+        mock_pyfunc_wrapper._strip_input_from_response_in_instruction_pipelines(
+            pipeline_input, pipeline_output, "generated_text"
+        )
+        == expected_output
+    )
+
+
+def test_invalid_instruction_pipeline_parsing(mock_pyfunc_wrapper):
+    prompt = "What is your favorite boba flavor?"
+
+    bad_output = {"generated_text": ["Strawberry Milk Cap", "Honeydew with boba"]}
+
+    with pytest.raises(MlflowException, match="Unable to parse the pipeline output. Expected"):
+        mock_pyfunc_wrapper._strip_input_from_response_in_instruction_pipelines(
+            prompt, bad_output, "generated_text"
+        )
+
+
+@pytest.mark.skipif(RUNNING_IN_GITHUB_ACTIONS, reason=GITHUB_ACTIONS_SKIP_REASON)
+@pytest.mark.skipcacheclean
```

Comment:
This test definitely cannot run on Github Actions (the size of the raw model weights is 3x the size of available RAM on a GHA runner)